### ISSUE TITLE
[BUG] [Linter] Linter doesn't raise H025 when tags in different blocks match

### ISSUE THREAD

**Issue by MrkGrgsn** [Riverside-Healthcare/djLint#787]
- [X] I'm on the [latest version](https://pypi.org/project/djlint/) of djLint
- [X] I've searched the [issues](https://github.com/Riverside-Healthcare/djLint/issues)
- [X] I've read the [docs](https://djlint.com)

## System Info

- OS: Fedora 38
- Python Version 3.11.6
- djLint Version 1.34.0
- template language: N/A

## Issue

The linter doesn't raise an H025 for start and end tags are in different blocks, ie, where the HTML is not well-formed. Examples below.

## How To Reproduce

No warning is raised for the `b` tags in this example.
```html
<h1>blah <b>bold</h1>
<p>blah</b> blah blah</p>
```

## Contents of .djlintrc/pyproject.toml [tool.djlint]

```yaml
{
    "profile": "django",
    "indent": 2,
    "ignore": "T003,H021"
}
```
